flexmeasures/data/models/planning/storage.py

```diff
@@ -172,7 +180,191 @@ def schedule(
             battery_schedule = ems_schedule[0]
 
         # Round schedule
-        if round_to_decimals:
-            battery_schedule = battery_schedule.round(round_to_decimals)
+        if self.round_to_decimals:
+            battery_schedule = battery_schedule.round(self.round_to_decimals)
 
         return battery_schedule
+
+    def persist_flex_model(self):
+        """Store new soc info as GenericAsset attributes"""
+        self.sensor.generic_asset.set_attribute("soc_datetime", self.start.isoformat())
+        if self.flex_model.get("soc_unit") == "kWh":
+            self.sensor.generic_asset.set_attribute(
+                "soc_in_mwh", self.flex_model["soc_at_start"] / 1000
+            )
+        else:
+            self.sensor.generic_asset.set_attribute(
+                "soc_in_mwh", self.flex_model["soc_at_start"]
+            )
+
+    def inspect_flex_config(self):
+        """
+        Check storage flex model and the flex context against schemas.
+        Before that, we fill in values from wider context, if possible. Mostly, we allow several fields to come from sensor attributes.
+
+        Note: Before we apply the flex model schema, we need to use the flex model identifiers with hyphens,
```

Comment:
The note also applies to *flex context*, so I'd change *flex model* to *flex* **config** (twice).